src/main/perf/LineFileDocs.java

```diff
@@ -745,28 +751,71 @@ public Document nextDoc(DocState doc) throws IOException {
     }
   }
 
-  private static class LineFileDoc {
-    final FloatBuffer vector;
-    final String stringText;
-    final ByteBuffer byteText;
+  private static abstract class LineFileDoc {
+    private final FloatBuffer vector;
 
-    LineFileDoc(String text, float[] vector) {
-      stringText = text;
-      byteText = null;
+    LineFileDoc(float[] vector) {
       if (vector == null) {
         this.vector = null;
       } else {
         this.vector = FloatBuffer.wrap(vector);
       }
     }
 
-    LineFileDoc(ByteBuffer bytes, float[] vector) {
-      stringText = null;
-      byteText = bytes;
-      if (vector == null) {
-        this.vector = null;
-      } else {
-        this.vector = FloatBuffer.wrap(vector);
+    abstract int getId();
+
+    String getStringText() {
+      throw new UnsupportedOperationException();
+    }
+
+    ByteBuffer getByteText() {
+      throw new UnsupportedOperationException();
+    }
+
+    private static final class TextBased extends LineFileDoc {
+      final String stringText;
+
+      final int id; // will be the exact id if this is txt based LFD
```

Comment:
Maybe update the comment?  Isn't it always a txt based LFD since this class is `TextBased`?